src/parser.cpp

```diff
@@ -2,6 +2,41 @@
 #include <string>
 #include <vector>
 
-Config parse(int argc, char *argv[], std::vector<std::string> input_lines) {
-  return Config(false, input_lines);
+bool validate(char *argv[]);
+
+absl::StatusOr<Config> parse(int argc, char *argv[], std::vector<std::string> input_lines) {
+
+    if(validate(argv))
+        return Config(false, input_lines);
+    else
+        return absl::InvalidArgumentError("invalid flag");
```

Comment:
We should provide error for an individual flag, e.g. - "`single` (boolean) - `<flag-value>` is not a valid input. Please see `--help`"